### PR TITLE
[PATCH v5] pkg-config and installcheck improvements

### PR DESCRIPTION
**Pull request by JereLeppanen**
Rework pkg-config files so that application can link libodp and libodphelper statically with pkg-config --static, without the -static linker option.

Don't run test suites in installcheck. Instead build and run a minimal test application. This saves a lot of time in distcheck, since tests are not run twice. It's also a better test, since currently installcheck doesn't build an application against the installed headers and libraries, but just re-runs the tests built earlier in make all.

Do installcheck in CI build jobs.

v2:
- Rebase.
- Add explanation in internal pc file.

v3:
- Add review tags.

v4:
- Rebase.
- Fix library name in test/miscellaneous/Makefile.am.

v5:
- Add review tags.

### CODE REVIEW

File: platform/linux-generic/libodp-linux.pc.in [hunk 1]
```diff
@@ -6,7 +6,6 @@ includedir=@includedir@
 Name: lib@ODP_LIB_NAME@
 Description: The ODP packet processing engine
 Version: @PKGCONFIG_VERSION@
-Requires.private: libconfig@AARCH64CRYPTO_PKG@
-Libs: -L${libdir} -l@ODP_LIB_NAME@ @ATOMIC_LIBS_NON_ABI_COMPAT@
-Libs.private: @OPENSSL_STATIC_LIBS@ @DPDK_LIBS@ @PCAP_LIBS@ @PTHREAD_LIBS@ @TIMER_LIBS@ @LIBXDP_LIBS@ -lpthread @ATOMIC_LIBS_ABI_COMPAT@
+Requires: lib@ODP_LIB_NAME@-internal
```
Comment (TuomasTaipale) [inline platform/linux-generic/libodp-linux.pc.in:9]: Perhaps a few words about how this structure (two separate pkg-config files with `Requires`) accomplishes a correct static linking outcome could be added to the commit log.

Comment (JereLeppanen) [reply platform/linux-generic/libodp-linux.pc.in:9]: Added explanation as a comment in the file itself. In principle, the same explanation belongs in odphelper-internal as well, but it's probably enough that it's in one place.

**Review by TuomasTaipale — APPROVED** [review]

File: test/miscellaneous/Makefile.am [hunk 2]
```diff
@@ -10,3 +10,35 @@ odp_api_from_cpp_SOURCES = odp_api_from_cpp.cpp
...
+
+if enable_shared
+
+PROGRAM_shared = odp_api_headers_shared
+
+installcheck-local: $(PROGRAM_shared)
+
+$(PROGRAM_shared): $(srcdir)/$(odp_api_headers_SOURCES)
+	$(CC) $(AM_CFLAGS) $(CFLAGS) $^ -o $@ \
+		`$(PKGCONFIG) libodphelper` `$(PKGCONFIG) libodp-linux`
```
Comment (MatiasElo) [inline test/miscellaneous/Makefile.am:24]: Could the ODP library name (`libodp-linux`) be passed using a variable? Now other platform will need to patch this makefile manually.

File: test/miscellaneous/Makefile.am [hunk 3]
```diff
@@ -10,3 +10,35 @@ odp_api_from_cpp_SOURCES = odp_api_from_cpp.cpp
...
+
+if enable_static
+
+PROGRAM_static = odp_api_headers_static
+
+installcheck-local: $(PROGRAM_static)
+
+$(PROGRAM_static): $(srcdir)/$(odp_api_headers_SOURCES)
+	$(CC) $(AM_CFLAGS) $(CFLAGS) $^ -o $@ \
+		`$(PKGCONFIG) --static libodphelper` `$(PKGCONFIG) --static libodp-linux`
```
Comment (MatiasElo) [inline test/miscellaneous/Makefile.am:43]: Same comment as above.

File: test/miscellaneous/Makefile.am (hunk 2 above)
Comment (JereLeppanen) [reply test/miscellaneous/Makefile.am]: Yes, that's how it was supposed to be. Good catch, I somehow missed that. Fixed in v4.

**Review by MatiasElo — APPROVED** [review]

**Pull request merged**